From 7dc8f4c04df41636a3b81f7c41c05d9adf78e768 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Fri, 21 Dec 2018 11:04:56 +0100
Subject: [PATCH 1/1] arm64: dts: marvell: mcbin: reserve PSCI area

The memory area [0x4000000-0x4200000[ is occupied by the PSCI firmware. Any
attempt to access it from Linux leads to an immediate crash.

So let's make the same memory reservation as the vendor kernel.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index 56fa44860909..18df1a321b52 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -23,6 +23,17 @@
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		psci_reserved: psci-area@4000000 {
+		reg = <0x0 0x4000000 0x0 0x200000>;
+		no-map;
+		};
+	};
+
 	aliases {
 		ethernet0 = &cp0_eth0;
 		ethernet1 = &cp1_eth0;
-- 
2.19.2

